app/controllers/application_controller.rb

```diff
@@ -1,10 +1,8 @@
 class ApplicationController < ActionController::API
 
-  rescue_from ActiveRecord::RecordNotFound, :with => :error_404
+  rescue_from ActiveRecord::RecordNotFound, with: ->{ show_errors(404) }
 
-  private
-
-  def error_404
-    head 404
+  def show_errors(status_code = params[:status_code] || 500)
+    head status_code
```

Comment:
agree. ideally this would be in its own `ErrorsController`. unfortunately,
there's no simple way to get `rescue_from` to reference an action from
`ErrorsController`. only option being, we'd have to duplicate behaviour.

On 24 March 2015 at 14:04, Alex Tomlins <notifications@github.com> wrote:

> In app/controllers/application_controller.rb
> <https://github.com/alphagov/url-arbiter/pull/21#discussion_r27009694>:
>
> >
> > -  private
> > -
> > -  def error_404
> > -    head 404
> > +  def show_errors(status_code = params[:status_code] || 500)
> > +    head status_code
>
> This will cause all controllers (that inherit from this) to have a
> show_errors action. Probably best avoided if possible.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/alphagov/url-arbiter/pull/21/files#r27009694>.
>



-- 
विनय
vinay